src/scipp/curve_fit.py

```diff
@@ -176,6 +178,70 @@ def _reshape_bounds(bounds):
     return left, right
 
 
+def _select_data_params_and_bounds(sel, da, p0, bounds):
+    dim, i = sel
+    return (
+        da[dim, i],
+        {k: v[dim, i] if dim in v.dims else v for k, v in p0.items()},
+        {
+            k: (
+                le[dim, i] if dim in le.dims else le,
+                ri[dim, i] if dim in ri.dims else ri,
+            )
+            for k, (le, ri) in bounds.items()
+        },
+    )
+
+
+def _serialize_variable(v):
+    return (v.dims, v.values, v.variances, str(v.unit) if v.unit is not None else None)
+
+
+def _serialize_mapping(v):
+    return (tuple(v.keys()), tuple(map(_serialize_variable, v.values())))
+
+
+def _serialize_bounds(v):
+    return (
+        tuple(v.keys()),
+        tuple(tuple(map(_serialize_variable, pair)) for pair in v.values()),
+    )
+
+
+def _serialize_data_array(da):
+    return (
+        _serialize_variable(da.data),
+        _serialize_mapping(da.coords),
+        _serialize_mapping(da.masks),
+    )
```

Comment:
Can't this use https://scipp.github.io/generated/modules/scipp.compat.dict.to_dict.html#scipp.compat.dict.to_dict?

Or maybe even https://github.com/scipp/scipp/blob/main/src/scipp/serialization.py, so `pickle` is not needed?

Reply:
A previous version of this PR used `save_hdf5` to a bytesIO stream and sent that. But it was slower than the current solution. I guess the hdf5 serialization is expensive?

I'll have a look at `compat.to_dict` and see if that can be used instead 👍

Reply:
Using the `to_dict` method ran into: `TypeError: cannot pickle 'scipp._scipp.core.Unit' object` and `TypeError: cannot pickle 'scipp._scipp.core.DType' object`

With some tweaks it would probably be made to work anyway, but I don't think the resulting code wouldn't be less than it is now. The bounds serialization would have to be custom anyway.